File: shopapp/shopapp/src/main/java/com/project/shopapp/dtos/OrderDto.java
package com.project.shopapp.dtos;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;
import java.math.BigDecimal;
import java.util.List;


@Setter
@Getter
@Data
@AllArgsConstructor
@NoArgsConstructor
public class OrderDto {
    @JsonProperty("full_name")
    private String fullName;
    private String email;
    @JsonProperty("phone_number")
    @Size(min = 9, message = "Số điện thoại phải có ít nhất 9 ký tự")
    @NotBlank(message = "Số điện thoại là bắt buộc")
    private String phoneNumber;
    @NotBlank(message = "Địa chỉ là bắt buộc")
    private String address;
    private String note;
    @JsonProperty("total_price")
    @Min(value = 0, message = "Tổng số tiền phải lớn hơn 0")
    private BigDecimal totalPrice;
    @JsonProperty("order_details")
    private List<OrderDetailDto> orderDetails;
    @JsonProperty("payment")
    private PaymentDto payment;
}

File: shopapp/shopapp_angular/src/app/dtos/order.dto.ts
import { OrderDetailDto } from "./order-detail.dto";
import { PaymentDto } from "./payment.dto";
export interface OrderDto {
  full_name: string;
  email: string;
  phone_number: string;
  note: string;
  address: string;
  address_detail:string
  order_details: OrderDetailDto[];
  payment: PaymentDto;
  is_buy_now: boolean;
  coupon_code: string;
  province: string,
  district: string,
  ward: string,
  shipping_method: string
}